src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs

```diff
@@ -0,0 +1,293 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression
+{
+    /// <summary>Provides methods and properties used to compress and decompress streams by using the ZLib data format specification.</summary>
+    public sealed class ZLibStream : Stream
```

Comment:
It almost seems like having a DelegatingStream class would be useful.

Reply:
Now you know why it was top of mind for me ;-)

That said, it wouldn't help much here, as this type would still need to override most of the methods.

And on top of that, it would expose what's effectively an implementation detail into the public API surface area.

Reply:
> That said, it wouldn't help much here, as this type would still need to override most of the methods.

why?

Reply:
> why?

Because it's maintaining the same semantics as GZipStream and has a ThrowIfClosed call at the beginning of most of the overrides before it calls to the delegated stream.

Reply:
Why wouldn't a DelegatingStream class do the same thing?

Reply:
There is a set of common stream modifiers that are useful that we should consider adding.  DelegatingStream, SubsettingStream, Readonly, WriteOnly, NoDispose.  I think I've written / seen most of these multiple times.  Do we have an issue?  cc @carlossanlop

Reply:
> Why wouldn't a DelegatingStream class do the same thing?

Seems to me like policy we wouldn't want there.  I could be wrong.

Reply:
But again, I think that's the lesser issue; making ZLibStream derive from DelegatingStream bakes that implementation detail into the public API surface area.  That seems to me very undesirable.  I'm very much in favor of adding a DelegatingStream for use in implementation, but I'm not sure how much of a place it has in public API surface area.

Reply:
> making ZLibStream derive from DelegatingStream bakes that implementation detail into the public API surface area. That seems to me very undesirable. 

I get the point, but it seems unfortunate. The broader implication here is that derivation to support code reuse is not a viable approach in our public API. That kinda sucks.